src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx

```diff
@@ -16,68 +15,55 @@ import { actions, selectors } from '../../../../../data/redux';
-        >
-          <FormattedMessage {...messages.cancelButtonLabel} />
-        </Button>
-        <Button
-          aria-label={intl.formatMessage(messages.selectButtonAriaLabel)}
-          onClick={hooks.onSelect({
-            selected,
-            updateField,
-            setBlockTitle,
-            defaultSettings,
-          })}
-          disabled={!selected}
-        >
-          <FormattedMessage {...messages.selectButtonLabel} />
-        </Button>
-      </ActionRow>
-    </ModalDialog.Footer>
-  </div>
-);
+}) => {
+  const intl = useIntl();
+  const defaultSettings = useSelector(selectors.problem.defaultSettings);
+  const dispatch = useDispatch();
+  const updateField = React.useCallback((data) => dispatch(actions.problem.updateField(data)), [dispatch]);
+  const setBlockTitle = React.useCallback((title) => dispatch(actions.app.setBlockTitle(title)), [dispatch]);
+  return (
+    <div className="editor-footer fixed-bottom">
+      <ModalDialog.Footer className="border-top-0">
+        <ActionRow>
+          <ActionRow.Spacer />
+          <Button
+            aria-label={intl.formatMessage(messages.cancelButtonAriaLabel)}
+            variant="tertiary"
+            onClick={onCancel}
+          >
+            <FormattedMessage {...messages.cancelButtonLabel} />
+          </Button>
+          <Button
+            aria-label={intl.formatMessage(messages.selectButtonAriaLabel)}
+            onClick={hooks.onSelect({
+              selected,
+              updateField,
+              setBlockTitle,
+              defaultSettings,
+            })}
+            disabled={!selected}
+          >
+            <FormattedMessage {...messages.selectButtonLabel} />
+          </Button>
+        </ActionRow>
+      </ModalDialog.Footer>
+    </div>
+  );
+};
 
 SelectTypeFooter.defaultProps = {
   selected: null,
 };
 
 SelectTypeFooter.propTypes = {
-  defaultSettings: PropTypes.shape({
-    maxAttempts: PropTypes.number,
-    rerandomize: PropTypes.string,
-    showResetButton: PropTypes.bool,
-    showanswer: PropTypes.string,
-  }).isRequired,
+  // defaultSettings: PropTypes.shape({
+  //   maxAttempts: PropTypes.number,
+  //   rerandomize: PropTypes.string,
+  //   showResetButton: PropTypes.bool,
+  //   showanswer: PropTypes.string,
+  // }).isRequired,
   onCancel: PropTypes.func.isRequired,
   selected: PropTypes.string,
-  updateField: PropTypes.func.isRequired,
-  setBlockTitle: PropTypes.func.isRequired,
-  // injected
-  intl: intlShape.isRequired,
-};
-
-export const mapStateToProps = (state) => ({
-  defaultSettings: selectors.problem.defaultSettings(state),
-});
-
-export const mapDispatchToProps = {
-  updateField: actions.problem.updateField,
-  setBlockTitle: actions.app.setBlockTitle,
 };
 
-export const SelectTypeFooterInternal = SelectTypeFooter; // For testing only
-export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(SelectTypeFooter));
+export default SelectTypeFooter;
```

Comment:
I modernized this component by:
* changing from `propTypes` to TypeScript `Props`
* Changing `injectIntl` to `useIntl`
* Changing `connect` to `useSelector` and `useDispatch`
* Testing it in an actual browser in `src/editors/containers/ProblemEditor/components/SelectTypeModal/index.test.tsx` rather than shallow snapshot testing

There is no change to the actual code within the component itself.